test/d2l-filter-dropdown/d2l-filter-dropdown.js

```diff
@@ -79,289 +79,317 @@ import { afterNextRender } from '@polymer/polymer/lib/utils/render-status.js';
+						assert.equal(pages[i].options.length, opts.length);
+						for (var j = 0; j < opts.length; j++) {
+							assert.equal (opts[j].selected, expected[i].options[j].selected);
+						}
 					}
-				}
-				done();
+					done();
+				});
+			});
+			test('adding duplicate filter option updates filters', function() {
+				const update = 2;
+				const newTitle = 'newTitle';
+				_setSelectedOptions([0, 4, 5]);
+				const expected = _getExpectedAndImport(filter);
+				filter.addFilterOption(options[update].cat, options[update].key, newTitle, true);
+				_updateExpectedOption(update, true, newTitle, expected);
+				assert.deepEqual(expected, filter._filters);
+			});
+			test('adding duplicate filter option updates display', function(done) {
+				const update = 2;
+				const newTitle = 'newTitle';
+				const selected = [0, 4, 5];
+				_setSelectedOptions(selected);
+				const expected = _getExpectedAndImport(filter);
+				filter.addFilterOption(options[update].cat, options[update].key, 'newTitle', true);
+				_updateExpectedOption(update, true, newTitle, expected);
+				window.requestAnimationFrame(function() {
+					var pages = _getPages();
+					for (var i = 0; i < pages.length; i++) {
+						var opts = _getOptions(pages[i]);
+						assert.equal(pages[i].options.length, opts.length);
+						for (var j = 0; j < opts.length; j++) {
+							assert.equal (opts[j].selected, expected[i].options[j].selected);
+							assert.equal(opts[j].text, expected[i].options[j].title);
+						}
+					}
+					done();
+				});
 			});
 		});
 	});
 	suite('d2l-filter-dropdown-page', function() {
 		setup(function() {
 			pageFixture = fixture('page');
 		});
-		test('contains returns true when substring exists', function() {
-			const found = pageFixture._caseInsensitiveContainsSubstring('Hello World', 'Hello');
-			assert.equal(true, found);
-		});
-		test('contains returns false when substring doesn\'t exist', function() {
-			const found = pageFixture._caseInsensitiveContainsSubstring('Hello World', 'HelloWorld');
-			assert.equal(false, found);
-		});
-		test('contains returns true when substring exists, regardless of case', function() {
-			const found = pageFixture._caseInsensitiveContainsSubstring('Hello World', 'hello');
-			assert.equal(true, found);
-		});
-		test('contains can handle special characters', function() {
-			const found = pageFixture._caseInsensitiveContainsSubstring('Hello World-/\\^$*+?.()|[]{}', 'world-/\\^$*+?.()|[]{}');
-			assert.equal(true, found);
+
+		test('searching emits event with correct context', (done) => {
+			const search = 'abc123';
+			const key = 'this is a key';
+			pageFixture.parentKey = key;
+
+			pageFixture.addEventListener('d2l-filter-dropdown-page-searched', (e) => {
+				expect(e.composed).to.be.true;
+				expect(e.bubbles).to.be.true;
+				expect(e.detail.value).to.equal(search);
+				expect(e.detail.categoryKey).to.equal(key);
+				done();
+			});
+
+			window.requestAnimationFrame(() => {
```

Comment:
I just copied what I saw from this file. Am I doing it right??

Reply:
This is basically asking for the next UI frame which allows some of the queued operations to proceed.  When people use `window.setTimeout` with a timeout of `0`, they probably wanted to use this.

You may not need this here though, since nothing you've done before this would need time to be applied.

Reply:
- [ ] make sure this is necessary